types/lodash/ts3.1/common/array.d.ts

```diff
@@ -0,0 +1,2120 @@
+    }
+    interface LoDashStatic {
+        /**
+         * Recursively flattens a nested array.
+         *
+         * @param array The array to recursively flatten.
+         * @return Returns the new flattened array.
+         */
+        flattenDeep<T>(array: ListOfRecursiveArraysOrValues<T> | null | undefined): T[];
+    }
+    interface Collection<T> {
+        /**
+         * @see _.flattenDeep
+         */
+        flattenDeep(): Collection<T>;
+    }
+    interface CollectionChain<T> {
+        /**
+         * @see _.flattenDeep
+         */
+        flattenDeep(): CollectionChain<T>;
+    }
+    interface LoDashStatic {
+        /**
+         * Recursively flatten array up to depth times.
+         *
+         * @param array The array to recursively flatten.
+         * @param number The maximum recursion depth.
+         * @return Returns the new flattened array.
+         */
+        flattenDepth<T>(array: ListOfRecursiveArraysOrValues<T> | null | undefined, depth?: number): T[];
+    }
+    interface Collection<T> {
+        /**
+         * @see _.flattenDepth
+         */
+        flattenDepth(depth?: number): Collection<T>;
+    }
+    interface CollectionChain<T> {
+        /**
+         * @see _.flattenDepth
+         */
+        flattenDepth(depth?: number): CollectionChain<T>;
+    }
+    interface LoDashStatic {
+        /**
+         * The inverse of `_.toPairs`; this method returns an object composed
+         * from key-value `pairs`.
+         *
+         * @category Array
+         * @param pairs The key-value pairs.
+         * @returns Returns the new object.
+         * @example
+         *
+         * _.fromPairs([['fred', 30], ['barney', 40]]);
+         * // => { 'fred': 30, 'barney': 40 }
+         */
+        fromPairs<T>(pairs: List<[PropertyName, T]> | null | undefined): Dictionary<T>;
+        /**
+         * @see _.fromPairs
+         */
+        fromPairs(pairs: List<any[]> | null | undefined): Dictionary<any>;
+    }
+    interface Collection<T> {
+        /**
+         * @see _.fromPairs
+         */
+        fromPairs(): Object<Dictionary<T extends [PropertyName, infer U] ? U : any>>;
+    }
+    interface CollectionChain<T> {
+        /**
+         * @see _.fromPairs
+         */
+        fromPairs(): ObjectChain<Dictionary<T extends [PropertyName, infer U] ? U : any>>;
+    }
+    interface LoDashStatic {
+        /**
+         * Gets the first element of array.
+         *
+         * @alias _.first
+         *
+         * @param array The array to query.
+         * @return Returns the first element of array.
+         */
+        head<T>(array: List<T> | null | undefined): T | undefined;
+    }
+    interface String {
+        /**
+         * @see _.head
+         */
+        head(): string | undefined;
+    }
+    interface StringChain {
+        /**
+         * @see _.head
+         */
+        head(): StringChain;
```

Comment:
I think this should also be StringNullableChain? Since `_.chain("").head() === undefined`

Reply:
Nice catch!